aten/src/ATen/native/Indexing.cpp

```diff
@@ -226,34 +231,192 @@ static std::tuple<Tensor, Tensor> makeLinearIndex(Tensor self, TensorList orig)
   return std::make_tuple(self, linearIndex);
 }
 
-Tensor index(const Tensor & self, TensorList indices) {
-  AT_CHECK(indices.size() <= (size_t)self.dim(),
-           "too many indices for tensor of dimension ", self.dim(), " (got ", indices.size(), ")");
+static bool all_strides_match(TensorList tensors) {
+  AT_ASSERT(tensors.size() >= 1);
+  auto strides = tensors[0].strides();
+  for (auto& tensor : tensors.slice(1)) {
+    if (!strides.equals(tensor.strides())) {
+      return false;
+    }
+  }
+  return true;
+}
+
+static std::string shapes_as_str(TensorList tensors) {
+  std::ostringstream os;
+  bool first = true;
+  for (auto& tensor : tensors) {
+    if (tensor.defined()) {
```

Comment:
What do undefined Tensors represent here? Dimensions where we take thw first element? Or range? Should they be mentionned here as otherwise the print migh have less indexers than dimensions (and shift them to the left)

Reply:
I believe undefined tensors are equivalent to the `:` in python

Reply:
So maybe we want to print something saying that we keep this dimension in full?

Reply:
Yes, it's equivalent to `:` in Python.

The non-indexed tensors aren't relevant here. For example, with:

```
res = self[idx_1, :, 2:5, ..., idx_2]
```

The only relevant condition is that the tensors `idx_1` and `idx_2` can be broadcast together. The other dimensions don't affect this part of the operation.